### ISSUE TITLE
[v3] Implement new Concept Exercise: basics

### ISSUE THREAD

**Issue by silvanocerza** [exercism/cpp#379]
This issue describes how to implement the `basics` concept exercise for the C++ track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student basic concepts of the C++ language.

## Learning objectives

- Know what a variable is.
- Know how to define a variable.
- Know how to update a variable.
- Know how to use type inference for variables.
- Know how to define a method.
- Know how to return a value from a method.
- Know how to call a method.
- Know that methods must be defined in classes.
- Know about the `public` access modifier.
- Know about the `static` modifier.
- Know how to define an integer.
- Know how to use mathematical operators on integers.
- Know how to define single- and multiline comments.

## Out of scope

- Naming rules for identifiers.
- Generic values.
- Memory and performance characteristics.
- Method overloads.
- Lambda's.
- Classes.
- Organizing methods in namespaces.
- Visibility.

## Concepts

- `basics`: know what a variable is; know how to define a variable; know how to update a variable; know how to use type inference for variables; know how to define a method; know how to return a value from a method; know how to call a method; know that methods must be defined in classes; know about the `public` access modifier; know about the `static` modifier; know how to defined an integer; know how to use mathematical operators on integers; know how to define single and multiline comments.

## Prerequisites

There are no prerequisites.

## Resources to refer to

None.

### Hints

- How to define an [integer][integers].
- How to define a [method][methods] without any arguments.
- How to [return][return] an integer from a function.
- How to add or subtract integers using [arithmetic operators][arithmetic-operators].
- How to [invoke][invocation] a function.

### After

- More information on the [`auto` keyword][auto].
- More info on the [assignment operators][assignment-operators].
- More info on [`class`es][class].
- Introduction of the [`static` keyword][static].
- More information on [scoping][scope].
- Difference between methods and [functions][function].
- More info on [commenting syntax][comment].

## Representer

This exercise does not require any specific representation logic to be added to the representer.

## Analyzer

This exercise does not require any specific analyzer logic to be added to the analyzer.

## Implementing

To implement this exercise, please follow [these instructions][implementation-instructions].

[integers]: https://en.cppreference.com/w/cpp/language/types#Integer_types
[methods]: https://en.cppreference.com/w/cpp/language/member_functions
[return]: https://en.cppreference.com/w/cpp/language/return
[arithmetic-operators]: https://en.cppreference.com/w/cpp/language/operator_arithmetic
[invocation]: https://en.cppreference.com/w/cpp/language/operator_other#Built-in_function_call_operator
[implementation-instructions]: https://github.com/exercism/v3/blob/master/languages/cpp/reference/implementing-a-concept-exercise.md
[auto]: https://en.cppreference.com/w/cpp/language/auto
[assignment-operators]: https://en.cppreference.com/w/cpp/language/operator_assignment
[class]: https://en.cppreference.com/w/cpp/language/class
[function]: https://en.cppreference.com/w/cpp/language/functions
[static]: https://en.cppreference.com/w/cpp/language/static
[scope]: https://en.cppreference.com/w/cpp/language/scope
[comments]: https://en.cppreference.com/w/cpp/comment

**Comment by vaeng** [comment]
There is a basic concept since 2023 that covers many of the aspects above.
